lib/liquid/parse_tree_visitor.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module Liquid
+  class ParseTreeVisitor
+    def self.for(node, callbacks = Hash.new(proc {}))
+      if defined?(node.class::ParseTreeVisitor)
```

Comment:
I feel like this subclassing pattern seems overkill. Why not just have each class define a `children` method (then you also wouldn't have to make so many accessors public) and then only have one `ParseTreeVisitor` class (rather than one subclass per tag)?

Reply:
As discussed above, using a full visitor pattern instead of a method that has to be public but that we don't want anyone to call seemed cleaner, and also allows us to keep traversal-related things out of the main namespace.

If you feel very strongly, I can change this back again.